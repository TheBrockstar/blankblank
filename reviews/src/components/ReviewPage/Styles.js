import styled from 'styled-components';
import { flexContainer } from '../../globalStyles/Styles';


// Review Page Components
export const ReviewContainer = styled.section`
  margin-top:  30px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

export const Rating = styled(flexContainer)`
  width: 200px;
  height: 200px;
  flex-direction: column;
  background-color: whitesmoke;
  border-radius: .7rem;
  background-color: rgba(53, 23, 6, 0.5);
  color: rgb(81,197,212);;
  font-weight: bold;
  margin-right: 1.5rem;
  align-self: center;

  span {
    font-size: 5.2rem;
  }


  @media only screen and (max-width: 450px) {
    width: 8rem;
    span {
      font-size: 3.3rem;
    }
  }
`;

export const Author = styled.h1`
  font-size: 2.5rem;
  font-weight: bold;
  margin-top: 30px;
  align-self: center;
  text-decoration: underline;+
`
export const ReviewDate = styled.p`
  font-size: 1.5rem;
  font-weight: bold;
  margin-top: 30px;
`
export const Quote = styled.blockquote`
  max-width: 600px;
  font-style: italic;
  font-size: 2.5rem;
  padding-left: 30px;
  border-left: 10px solid rgb(105,68,37);
  margin: 30px 0;
`

export const Stars = styled.div`
  font-size: 1.5rem;
  white-space: nowrap;
  overflow-x: hidden;
  width: ${props => ((props.rating / 5) * 100) + "%"} // Calculates the number of stars to show.

  @media only screen and (max-width: 450px) {
    font-size: .7rem;
  }
`;

export const ReturnToList = styled.button`
  width: 50%;
  border: none;
  background: none;
  cursor: pointer;
  margin-top: 60px;
  font-size: 1.5rem;
  color: rgb(53, 23, 6);
  align-self: center;
  border: 5px solid rgba(53, 23, 6, 0.5);
  border-radius: 10px;
  padding: 10px 0;

  &:hover {
    background-color: rgb(81,197,212);
    color: whitesmoke;
    border: 5px solid rgb(105,68,37);
  }
`