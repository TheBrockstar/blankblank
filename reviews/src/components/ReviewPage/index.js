import React, { useState, useEffect } from 'react';
import axios from 'axios';

import { ReviewContainer, Rating, Stars, Quote, Author, ReviewDate, ReturnToList } from './Styles';

import { Loading, NotFound404 } from '../../globalStyles/Styles';

function ReviewPage(props) {

  const [review, setReview] = useState(null);

  useEffect(() => {
  async function fetchReview() {
    try {
      const response = await axios(
      `https://shakespeare.podium.com/api/reviews/${props.match.params.id}`, 
      { 'headers': { 'x-api-key':'H3TM28wjL8R4#HTnqk?c' }}
      );

      setReview(response.data);
    } catch ( error ) {
      setReview('404')
    }
  }

  fetchReview();

 ;
  }, []);
    if (review) {
      return review === "404" ? (
        <NotFound404>
          404 - Not Found 
          <br/> 
          Get Thee Hence!
        </NotFound404>
      ) : (
        <ReviewContainer>
          <Rating>
            <span>{ review.rating }</span>
            <div>
              <Stars rating={ review.rating }>★ ★ ★ ★ ★</Stars>
            </div>
          </Rating>
          <Author>{ review.author }</Author>
          <Quote>“{ review.body }”</Quote>
          <ReviewDate>Reviewed On: { (new Date(review.publish_date)).toDateString() }</ReviewDate>
          <ReturnToList role='link' onClick={_ => props.history.push('/')}>Return to Full List</ReturnToList>
        </ReviewContainer>
      );
    } else {
      return (<Loading>Loading...</Loading>)
    }
    
}

export default ReviewPage;
