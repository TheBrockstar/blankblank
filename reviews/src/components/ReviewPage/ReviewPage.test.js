import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ReviewPage from './index';

describe ('ReviewPage', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ReviewPage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    expect(shallow(<ReviewPage />)).toMatchSnapshot();
  });

});