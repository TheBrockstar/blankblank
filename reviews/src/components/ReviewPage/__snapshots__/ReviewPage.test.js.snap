// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewPage matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Loading...",
    },
    "ref": null,
    "rendered": "Loading...",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": true,
        "lastClassName": "bqVEJT",
        "rules": Array [
          "
  font-size: 2rem;
  margin-top: 10rem;
  color: whitesmoke;
",
        ],
      },
      "displayName": "styled.h1",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "h1",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Loading...",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "lastClassName": "bqVEJT",
          "rules": Array [
            "
  font-size: 2rem;
  margin-top: 10rem;
  color: whitesmoke;
",
          ],
        },
        "displayName": "styled.h1",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "h1",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
